pkg/generate/app/cmd/newapp.go

```diff
@@ -189,24 +194,29 @@ func (c *AppConfig) ensureDockerSearch() {
 }
 
 // SetOpenShiftClient sets the passed OpenShift client in the application configuration
-func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespace string, dockerclient *docker.Client) {
-	c.OSClient = osclient
+func (c *AppConfig) SetOpenShiftClient(imageClient imageclient.ImageInterface, templateClient templateclient.TemplateInterface, routeClient routeclient.RouteInterface, OriginNamespace string, dockerclient *docker.Client) {
 	c.OriginNamespace = OriginNamespace
 	namespaces := []string{OriginNamespace}
 	if openshiftNamespace := "openshift"; OriginNamespace != openshiftNamespace {
 		namespaces = append(namespaces, openshiftNamespace)
 	}
+	c.ImageClient = imageClient
+	c.RouteClient = routeClient
+	c.TemplateClient = templateClient
 	c.ImageStreamSearcher = app.ImageStreamSearcher{
-		Client:            osclient,
-		ImageStreamImages: osclient,
+		Client:            c.ImageClient,
+		ImageStreamImages: c.ImageClient,
 		Namespaces:        namespaces,
 		AllowMissingTags:  c.AllowMissingImageStreamTags,
 	}
-	c.ImageStreamByAnnotationSearcher = app.NewImageStreamByAnnotationSearcher(osclient, osclient, namespaces)
+	c.ImageStreamByAnnotationSearcher = app.NewImageStreamByAnnotationSearcher(
+		c.ImageClient,
+		c.ImageClient,
```

Comment:
I'm not sure I wanna ask, but srsly? We need twice the same thing?

Reply:
> I'm not sure I wanna ask, but srsly? We need twice the same thing?

two different interfaces for getters.